[Automation] javac compile fails for jakarta.ws.rs:jakarta.ws.rs-api:3.0.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25469529742/job/74730397489
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
      ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:20: error: cannot find symbol
          GenericType<List<String>[]> genericType = new GenericType<List<String>[]>() {
                                                        ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:24: error: cannot find symbol
      private static final String CLIENT_BUILDER_PROPERTY = ClientBuilder.JAXRS_DEFAULT_CLIENT_BUILDER_PROPERTY;
                                                            ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:102: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:107: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:112: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:117: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:122: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:127: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:100: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:105: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:110: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:115: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:120: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:125: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:130: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:135: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:140: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:145: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:150: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:155: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:160: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:165: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:170: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:175: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:180: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:19: error: cannot find symbol
  public class ExtFactoryFinderTest extends RuntimeDelegate {
                                            ^
    symbol: class RuntimeDelegate
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:23: error: cannot find symbol
  public class FactoryFinderTest extends ClientBuilder {
                                         ^
    symbol: class ClientBuilder
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:9: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Application;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:10: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Link;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:11: error: package javax.ws.rs.core.Response does not exist
  import javax.ws.rs.core.Response.ResponseBuilder;
                                  ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:12: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.UriBuilder;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:13: error: package javax.ws.rs.core.Variant does not exist
  import javax.ws.rs.core.Variant.VariantListBuilder;
                                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:14: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:128: error: package Link does not exist
      public Link.Builder createLinkBuilder() {
                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:9: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java:9: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:11: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.GenericType;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:16: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.Client;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:17: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:18: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Configuration;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java:9: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:10: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.WebTarget;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:11: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:16: error: package SseEventSource does not exist
  public class SseFactoryFinderTest extends SseEventSource.Builder {
                                                          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:95: error: package SseEventSource does not exist
      protected SseEventSource.Builder target(WebTarget endpoint) {
                              ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:100: error: package SseEventSource does not exist
      public SseEventSource.Builder reconnectingEvery(long delay, TimeUnit unit) {
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:16: error: package SseEventSource does not exist
              SseEventSource.Builder.JAXRS_DEFAULT_SSE_BUILDER_PROPERTY;
                            ^
  100 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-07T00:52:39Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25469529742/job/74730397502
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
      ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:20: error: cannot find symbol
          GenericType<List<String>[]> genericType = new GenericType<List<String>[]>() {
                                                        ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:24: error: cannot find symbol
      private static final String CLIENT_BUILDER_PROPERTY = ClientBuilder.JAXRS_DEFAULT_CLIENT_BUILDER_PROPERTY;
                                                            ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:102: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:107: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:112: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:117: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:122: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:127: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:100: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:105: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:110: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:115: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:120: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:125: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:130: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:135: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:140: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:145: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:150: error: method does not override or implement a method from a supertype
      @Override

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:155: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:160: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:165: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:170: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:175: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:180: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:19: error: cannot find symbol
  public class ExtFactoryFinderTest extends RuntimeDelegate {
                                            ^
    symbol: class RuntimeDelegate
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:23: error: cannot find symbol
  public class FactoryFinderTest extends ClientBuilder {
                                         ^
    symbol: class ClientBuilder
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:9: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Application;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:10: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Link;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:11: error: package javax.ws.rs.core.Response does not exist
  import javax.ws.rs.core.Response.ResponseBuilder;
                                  ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:12: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.UriBuilder;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:13: error: package javax.ws.rs.core.Variant does not exist
  import javax.ws.rs.core.Variant.VariantListBuilder;
                                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:14: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:128: error: package Link does not exist
      public Link.Builder createLinkBuilder() {
                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:9: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java:9: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:11: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.GenericType;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:16: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.Client;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:17: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:18: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Configuration;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java:9: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:10: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.WebTarget;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:11: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:16: error: package SseEventSource does not exist
  public class SseFactoryFinderTest extends SseEventSource.Builder {
                                                          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:95: error: package SseEventSource does not exist
      protected SseEventSource.Builder target(WebTarget endpoint) {
                              ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:100: error: package SseEventSource does not exist
      public SseEventSource.Builder reconnectingEvery(long delay, TimeUnit unit) {
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:16: error: package SseEventSource does not exist
              SseEventSource.Builder.JAXRS_DEFAULT_SSE_BUILDER_PROPERTY;
                            ^
  100 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-07T00:52:36Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25469529742/job/74730397535
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
     ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:20: error: cannot find symbol
          GenericType<List<String>[]> genericType = new GenericType<List<String>[]>() {
                                                        ^
    symbol:   class GenericType
    location: class GenericTypeTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:24: error: cannot find symbol
      private static final String CLIENT_BUILDER_PROPERTY = ClientBuilder.JAXRS_DEFAULT_CLIENT_BUILDER_PROPERTY;
                                                            ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:36: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:54: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
              ^
    symbol:   class ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:72: error: cannot find symbol
              ClientBuilder builder = ClientBuilder.newBuilder();
                                      ^
    symbol:   variable ClientBuilder
    location: class FactoryFinderTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:102: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:107: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:112: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:117: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:122: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:127: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:100: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:105: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:110: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:115: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:120: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:125: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:130: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:135: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:140: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:145: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:150: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:155: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:160: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:165: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:170: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:175: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:180: error: method does not override or implement a method from a supertype
      @Override
      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:19: error: cannot find symbol
  public class ExtFactoryFinderTest extends RuntimeDelegate {
                                            ^
    symbol: class RuntimeDelegate
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:23: error: cannot find symbol
  public class FactoryFinderTest extends ClientBuilder {
                                         ^
    symbol: class ClientBuilder
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:9: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Application;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:10: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Link;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:11: error: package javax.ws.rs.core.Response does not exist
  import javax.ws.rs.core.Response.ResponseBuilder;
                                  ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:12: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.UriBuilder;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:13: error: package javax.ws.rs.core.Variant does not exist
  import javax.ws.rs.core.Variant.VariantListBuilder;
                                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:14: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java:128: error: package Link does not exist
      public Link.Builder createLinkBuilder() {
                 ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:9: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java:9: error: package javax.ws.rs.ext does not exist
  import javax.ws.rs.ext.RuntimeDelegate;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java:11: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.GenericType;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:16: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.Client;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:17: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java:18: error: package javax.ws.rs.core does not exist
  import javax.ws.rs.core.Configuration;
                         ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java:9: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.ClientBuilder;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:10: error: package javax.ws.rs.client does not exist
  import javax.ws.rs.client.WebTarget;
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:11: error: package javax.ws.rs.sse does not exist
  import javax.ws.rs.sse.SseEventSource;
                        ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:16: error: package SseEventSource does not exist
  public class SseFactoryFinderTest extends SseEventSource.Builder {
                                                          ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:95: error: package SseEventSource does not exist
      protected SseEventSource.Builder target(WebTarget endpoint) {
                              ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java:100: error: package SseEventSource does not exist
      public SseEventSource.Builder reconnectingEvery(long delay, TimeUnit unit) {
                           ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/2.1.6/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java:16: error: package SseEventSource does not exist
              SseEventSource.Builder.JAXRS_DEFAULT_SSE_BUILDER_PROPERTY;
                            ^
  100 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.0.0 at 2026-05-07T00:52:45Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 3.0.0 failed with exit code 1.
FAILED[javac compile][3.0.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0.0" ./gradlew clean compileTestJava -Pcoordinates="jakarta.ws.rs:jakarta.ws.rs-api:2.1.6"]
```

--- a/metadata/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/reachability-metadata.json
+++ b/metadata/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/reachability-metadata.json
@@ -1,0 +1,40 @@
+{
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.client.FactoryFinder"
+      },
+      "glob": "META-INF/services/jakarta.ws.rs.client.ClientBuilder"
+    },
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.client.ClientBuilder"
+      },
+      "glob": "jakarta/ws/rs/client/ClientBuilder.class"
+    },
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.ext.FactoryFinder"
+      },
+      "glob": "META-INF/services/jakarta.ws.rs.ext.RuntimeDelegate"
+    },
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.ext.RuntimeDelegate"
+      },
+      "glob": "jakarta/ws/rs/ext/RuntimeDelegate.class"
+    },
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.sse.FactoryFinder"
+      },
+      "glob": "META-INF/services/jakarta.ws.rs.sse.SseEventSource$Builder"
+    },
+    {
+      "condition": {
+        "typeReached": "jakarta.ws.rs.sse.SseEventSource$Builder"
+      },
+      "glob": "jakarta/ws/rs/sse/SseEventSource$Builder.class"
+    }
+  ]
+}

--- a/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
+++ b/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
@@ -25,7 +25,7 @@
       "3.0.0"
     ],
     "allowed-packages" : [
-      "javax.ws.rs"
+      "jakarta.ws.rs"
     ]
   }
 ]

--- a/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
+++ b/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
@@ -13,5 +13,14 @@
     "allowed-packages" : [
       "javax.ws.rs"
     ]
+  },
+  {
+    "metadata-version" : "3.0.0",
+    "tested-versions" : [
+      "3.0.0"
+    ],
+    "allowed-packages" : [
+      "javax.ws.rs"
+    ]
   }
 ]

--- a/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
+++ b/metadata/jakarta.ws.rs/jakarta.ws.rs-api/index.json
@@ -16,6 +16,11 @@
   },
   {
     "metadata-version" : "3.0.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/jakarta/ws/rs/jakarta.ws.rs-api/$version$/jakarta.ws.rs-api-$version$-sources.jar",
+    "repository-url" : "https://github.com/jakartaee/rest",
+    "test-code-url" : "https://github.com/jakartaee/rest/tree/$version$/jaxrs-api/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/jakarta/ws/rs/jakarta.ws.rs-api/$version$/jakarta.ws.rs-api-$version$-javadoc.jar",
+    "description" : "Jakarta RESTful Web Services API defines the standard annotations, interfaces, and runtime contracts for building RESTful services and clients in Jakarta EE. This artifact provides only the API types for resources, request and response handling, media negotiation, filters, interceptors, and extension points while implementations are supplied by Jakarta REST runtimes.",
     "tested-versions" : [
       "3.0.0"
     ],

--- a/stats/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/execution-metrics.json
+++ b/stats/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/execution-metrics.json
@@ -1,0 +1,109 @@
+{
+  "fix_javac_fail:2026-05-07": {
+    "timestamp": "2026-05-07T07:12:06.549484Z",
+    "library": "jakarta.ws.rs:jakarta.ws.rs-api:3.0.0",
+    "starting_commit": "b74a40fd7be563fe149ea1a53e22b3f74a985582",
+    "ending_commit": "4cc71b7af4e595d9f0067a45a9b129327747006e",
+    "previous_library": "jakarta.ws.rs:jakarta.ws.rs-api:2.1.6",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.0.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 16,
+            "totalCalls": 16
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 6,
+            "totalCalls": 6
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 22,
+        "totalCalls": 22
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 715,
+          "missed": 5618,
+          "ratio": 0.112901,
+          "total": 6333
+        },
+        "line": {
+          "covered": 185,
+          "missed": 1228,
+          "ratio": 0.130927,
+          "total": 1413
+        },
+        "method": {
+          "covered": 31,
+          "missed": 410,
+          "ratio": 0.070295,
+          "total": 441
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.1.6",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 16,
+            "totalCalls": 16
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 6,
+            "totalCalls": 6
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 22,
+        "totalCalls": 22
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 718,
+          "missed": 5677,
+          "ratio": 0.112275,
+          "total": 6395
+        },
+        "line": {
+          "covered": 186,
+          "missed": 1236,
+          "ratio": 0.130802,
+          "total": 1422
+        },
+        "method": {
+          "covered": 31,
+          "missed": 414,
+          "ratio": 0.069663,
+          "total": 445
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 55906,
+      "cached_input_tokens_used": 565760,
+      "output_tokens_used": 2587,
+      "iterations": 1,
+      "cost_usd": 0.3605,
+      "tested_library_loc": 185,
+      "metadata_entries": 6,
+      "previous_library_metadata_entries": 6,
+      "code_coverage_percent": 13.09,
+      "previous_library_coverage_percent": 13.08
+    },
+    "artifacts": {
+      "test_file": "tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java",
+      "metadata_file": "metadata/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/stats.json
+++ b/stats/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "3.0.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 16,
+          "totalCalls" : 16
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 6,
+          "totalCalls" : 6
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 22,
+      "totalCalls" : 22
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 715,
+        "missed" : 5618,
+        "ratio" : 0.112901,
+        "total" : 6333
+      },
+      "line" : {
+        "covered" : 185,
+        "missed" : 1228,
+        "ratio" : 0.130927,
+        "total" : 1413
+      },
+      "method" : {
+        "covered" : 31,
+        "missed" : 410,
+        "ratio" : 0.070295,
+        "total" : 441
+      }
+    }
+  } ]
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/.gitignore
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/build.gradle
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "jakarta.ws.rs:jakarta.ws.rs-api:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/gradle.properties
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = jakarta.ws.rs:jakarta.ws.rs-api:3.0.0
+library.version = 3.0.0
+metadata.dir = jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/settings.gradle
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'jakarta.ws.rs.jakarta.ws.rs-api_tests'

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java
@@ -1,0 +1,49 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import javax.ws.rs.client.ClientBuilder;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class ClientBuilderTest {
+    private static final String CLIENT_BUILDER_PROPERTY = ClientBuilder.JAXRS_DEFAULT_CLIENT_BUILDER_PROPERTY;
+    private static final String INVALID_CLIENT_BUILDER_CLASS = InvalidClientBuilderProvider.class.getName();
+
+    @Test
+    public void newBuilderReportsProviderLoadedAsWrongType() {
+        String previousProvider = System.getProperty(CLIENT_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, INVALID_CLIENT_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(ClientBuilderTest.class.getClassLoader());
+
+            assertThatThrownBy(ClientBuilder::newBuilder)
+                    .isInstanceOf(LinkageError.class)
+                    .hasMessageContaining("ClassCastException: attempting to cast")
+                    .hasMessageContaining("javax/ws/rs/client/ClientBuilder.class");
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(CLIENT_BUILDER_PROPERTY);
+        } else {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, previousProvider);
+        }
+    }
+
+    public static final class InvalidClientBuilderProvider {
+        public InvalidClientBuilderProvider() {
+        }
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ClientBuilderTest.java
@@ -6,7 +6,7 @@
  */
 package jakarta_ws_rs.jakarta_ws_rs_api;
 
-import javax.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.ClientBuilder;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
@@ -27,7 +27,7 @@ public class ClientBuilderTest {
             assertThatThrownBy(ClientBuilder::newBuilder)
                     .isInstanceOf(LinkageError.class)
                     .hasMessageContaining("ClassCastException: attempting to cast")
-                    .hasMessageContaining("javax/ws/rs/client/ClientBuilder.class");
+                    .hasMessageContaining("jakarta/ws/rs/client/ClientBuilder.class");
         } finally {
             restoreProviderProperty(previousProvider);
             Thread.currentThread().setContextClassLoader(previousContextClassLoader);

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java
@@ -1,0 +1,131 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.Response.ResponseBuilder;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.Variant.VariantListBuilder;
+import javax.ws.rs.ext.RuntimeDelegate;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ExtFactoryFinderTest extends RuntimeDelegate {
+    private static final String RUNTIME_DELEGATE_PROPERTY = RuntimeDelegate.JAXRS_RUNTIME_DELEGATE_PROPERTY;
+    private static final String TEST_RUNTIME_DELEGATE_CLASS = ExtFactoryFinderTest.class.getName();
+
+    @Test
+    public void getInstanceInstantiatesProviderWithCurrentClassLoaderWhenContextClassLoaderIsNull() {
+        String previousProvider = System.getProperty(RUNTIME_DELEGATE_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            RuntimeDelegate.setInstance(null);
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, TEST_RUNTIME_DELEGATE_CLASS);
+            Thread.currentThread().setContextClassLoader(null);
+
+            RuntimeDelegate delegate = RuntimeDelegate.getInstance();
+
+            assertThat(delegate).isInstanceOf(ExtFactoryFinderTest.class);
+        } finally {
+            RuntimeDelegate.setInstance(null);
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void getInstanceInstantiatesProviderWithContextClassLoader() {
+        String previousProvider = System.getProperty(RUNTIME_DELEGATE_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            RuntimeDelegate.setInstance(null);
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, TEST_RUNTIME_DELEGATE_CLASS);
+            Thread.currentThread().setContextClassLoader(ExtFactoryFinderTest.class.getClassLoader());
+
+            RuntimeDelegate delegate = RuntimeDelegate.getInstance();
+
+            assertThat(delegate).isInstanceOf(ExtFactoryFinderTest.class);
+        } finally {
+            RuntimeDelegate.setInstance(null);
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void getInstanceRetriesWithCurrentClassLoaderWhenContextClassLoaderCannotLoadProvider() {
+        String previousProvider = System.getProperty(RUNTIME_DELEGATE_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            RuntimeDelegate.setInstance(null);
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, TEST_RUNTIME_DELEGATE_CLASS);
+            Thread.currentThread().setContextClassLoader(new RejectingClassLoader());
+
+            RuntimeDelegate delegate = RuntimeDelegate.getInstance();
+
+            assertThat(delegate).isInstanceOf(ExtFactoryFinderTest.class);
+        } finally {
+            RuntimeDelegate.setInstance(null);
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(RUNTIME_DELEGATE_PROPERTY);
+        } else {
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, previousProvider);
+        }
+    }
+
+    private static final class RejectingClassLoader extends ClassLoader {
+        private RejectingClassLoader() {
+            super(null);
+        }
+
+        @Override
+        protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+            throw new ClassNotFoundException(name);
+        }
+    }
+
+    @Override
+    public UriBuilder createUriBuilder() {
+        return null;
+    }
+
+    @Override
+    public ResponseBuilder createResponseBuilder() {
+        return null;
+    }
+
+    @Override
+    public VariantListBuilder createVariantListBuilder() {
+        return null;
+    }
+
+    @Override
+    public <T> T createEndpoint(Application application, Class<T> endpointType) {
+        return null;
+    }
+
+    @Override
+    public <T> HeaderDelegate<T> createHeaderDelegate(Class<T> type) {
+        return null;
+    }
+
+    @Override
+    public Link.Builder createLinkBuilder() {
+        return null;
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/ExtFactoryFinderTest.java
@@ -6,12 +6,12 @@
  */
 package jakarta_ws_rs.jakarta_ws_rs_api;
 
-import javax.ws.rs.core.Application;
-import javax.ws.rs.core.Link;
-import javax.ws.rs.core.Response.ResponseBuilder;
-import javax.ws.rs.core.UriBuilder;
-import javax.ws.rs.core.Variant.VariantListBuilder;
-import javax.ws.rs.ext.RuntimeDelegate;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Link;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
+import jakarta.ws.rs.core.UriBuilder;
+import jakarta.ws.rs.core.Variant.VariantListBuilder;
+import jakarta.ws.rs.ext.RuntimeDelegate;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java
@@ -1,0 +1,199 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import java.security.KeyStore;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Configuration;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FactoryFinderTest extends ClientBuilder {
+    private static final String CLIENT_BUILDER_PROPERTY = ClientBuilder.JAXRS_DEFAULT_CLIENT_BUILDER_PROPERTY;
+    private static final String TEST_CLIENT_BUILDER_CLASS = FactoryFinderTest.class.getName();
+
+    @Test
+    public void newBuilderInstantiatesProviderWithCurrentClassLoaderWhenContextClassLoaderIsNull() {
+        String previousProvider = System.getProperty(CLIENT_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, TEST_CLIENT_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(null);
+
+            ClientBuilder builder = ClientBuilder.newBuilder();
+
+            assertThat(builder).isInstanceOf(FactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void newBuilderInstantiatesProviderWithContextClassLoader() {
+        String previousProvider = System.getProperty(CLIENT_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, TEST_CLIENT_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(FactoryFinderTest.class.getClassLoader());
+
+            ClientBuilder builder = ClientBuilder.newBuilder();
+
+            assertThat(builder).isInstanceOf(FactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void newBuilderRetriesWithCurrentClassLoaderWhenContextClassLoaderCannotLoadProvider() {
+        String previousProvider = System.getProperty(CLIENT_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, TEST_CLIENT_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(new RejectingClassLoader());
+
+            ClientBuilder builder = ClientBuilder.newBuilder();
+
+            assertThat(builder).isInstanceOf(FactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(CLIENT_BUILDER_PROPERTY);
+        } else {
+            System.setProperty(CLIENT_BUILDER_PROPERTY, previousProvider);
+        }
+    }
+
+    private static final class RejectingClassLoader extends ClassLoader {
+        private RejectingClassLoader() {
+            super(null);
+        }
+
+        @Override
+        protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+            throw new ClassNotFoundException(name);
+        }
+    }
+
+    @Override
+    public Configuration getConfiguration() {
+        return null;
+    }
+
+    @Override
+    public ClientBuilder property(String name, Object value) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Class<?> componentClass) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Class<?> componentClass, int priority) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Class<?> componentClass, Class<?>... contracts) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Class<?> componentClass, Map<Class<?>, Integer> contracts) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Object component) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Object component, int priority) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Object component, Class<?>... contracts) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder register(Object component, Map<Class<?>, Integer> contracts) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder withConfig(Configuration config) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder sslContext(SSLContext sslContext) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder keyStore(KeyStore keyStore, char[] password) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder trustStore(KeyStore trustStore) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder hostnameVerifier(HostnameVerifier verifier) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder executorService(ExecutorService executorService) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder scheduledExecutorService(ScheduledExecutorService scheduledExecutorService) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder connectTimeout(long timeout, TimeUnit unit) {
+        return this;
+    }
+
+    @Override
+    public ClientBuilder readTimeout(long timeout, TimeUnit unit) {
+        return this;
+    }
+
+    @Override
+    public Client build() {
+        return null;
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/FactoryFinderTest.java
@@ -13,9 +13,9 @@ import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import javax.net.ssl.HostnameVerifier;
 import javax.net.ssl.SSLContext;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
-import javax.ws.rs.core.Configuration;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Configuration;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java
@@ -1,0 +1,26 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import java.util.List;
+
+import javax.ws.rs.core.GenericType;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GenericTypeTest {
+    @Test
+    public void resolvesGenericArrayRawType() {
+        GenericType<List<String>[]> genericType = new GenericType<List<String>[]>() {
+        };
+
+        assertThat(genericType.getRawType()).isEqualTo(List[].class);
+        assertThat(genericType.getType().getTypeName()).contains("java.util.List<java.lang.String>[]");
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/GenericTypeTest.java
@@ -8,7 +8,7 @@ package jakarta_ws_rs.jakarta_ws_rs_api;
 
 import java.util.List;
 
-import javax.ws.rs.core.GenericType;
+import jakarta.ws.rs.core.GenericType;
 
 import org.junit.jupiter.api.Test;
 

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java
@@ -1,0 +1,51 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import javax.ws.rs.ext.RuntimeDelegate;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class RuntimeDelegateTest {
+    private static final String RUNTIME_DELEGATE_PROPERTY = RuntimeDelegate.JAXRS_RUNTIME_DELEGATE_PROPERTY;
+    private static final String INVALID_RUNTIME_DELEGATE_CLASS = InvalidRuntimeDelegateProvider.class.getName();
+
+    @Test
+    public void getInstanceReportsProviderLoadedAsWrongType() {
+        String previousProvider = System.getProperty(RUNTIME_DELEGATE_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            RuntimeDelegate.setInstance(null);
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, INVALID_RUNTIME_DELEGATE_CLASS);
+            Thread.currentThread().setContextClassLoader(RuntimeDelegateTest.class.getClassLoader());
+
+            assertThatThrownBy(RuntimeDelegate::getInstance)
+                    .isInstanceOf(LinkageError.class)
+                    .hasMessageContaining("ClassCastException: attempting to cast")
+                    .hasMessageContaining("javax/ws/rs/ext/RuntimeDelegate.class");
+        } finally {
+            RuntimeDelegate.setInstance(null);
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(RUNTIME_DELEGATE_PROPERTY);
+        } else {
+            System.setProperty(RUNTIME_DELEGATE_PROPERTY, previousProvider);
+        }
+    }
+
+    public static final class InvalidRuntimeDelegateProvider {
+        public InvalidRuntimeDelegateProvider() {
+        }
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/RuntimeDelegateTest.java
@@ -6,7 +6,7 @@
  */
 package jakarta_ws_rs.jakarta_ws_rs_api;
 
-import javax.ws.rs.ext.RuntimeDelegate;
+import jakarta.ws.rs.ext.RuntimeDelegate;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
@@ -28,7 +28,7 @@ public class RuntimeDelegateTest {
             assertThatThrownBy(RuntimeDelegate::getInstance)
                     .isInstanceOf(LinkageError.class)
                     .hasMessageContaining("ClassCastException: attempting to cast")
-                    .hasMessageContaining("javax/ws/rs/ext/RuntimeDelegate.class");
+                    .hasMessageContaining("jakarta/ws/rs/ext/RuntimeDelegate.class");
         } finally {
             RuntimeDelegate.setInstance(null);
             restoreProviderProperty(previousProvider);

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java
@@ -1,0 +1,50 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import javax.ws.rs.sse.SseEventSource;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SseEventSourceInnerBuilderTest {
+    private static final String SSE_BUILDER_PROPERTY =
+            SseEventSource.Builder.JAXRS_DEFAULT_SSE_BUILDER_PROPERTY;
+    private static final String INVALID_SSE_BUILDER_CLASS = InvalidSseEventSourceBuilderProvider.class.getName();
+
+    @Test
+    public void targetReportsProviderLoadedAsWrongType() {
+        String previousProvider = System.getProperty(SSE_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(SSE_BUILDER_PROPERTY, INVALID_SSE_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(SseEventSourceInnerBuilderTest.class.getClassLoader());
+
+            assertThatThrownBy(() -> SseEventSource.target(null))
+                    .isInstanceOf(LinkageError.class)
+                    .hasMessageContaining("ClassCastException: attempting to cast")
+                    .hasMessageContaining("javax/ws/rs/sse/SseEventSource$Builder.class");
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(SSE_BUILDER_PROPERTY);
+        } else {
+            System.setProperty(SSE_BUILDER_PROPERTY, previousProvider);
+        }
+    }
+
+    public static final class InvalidSseEventSourceBuilderProvider {
+        public InvalidSseEventSourceBuilderProvider() {
+        }
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseEventSourceInnerBuilderTest.java
@@ -6,7 +6,7 @@
  */
 package jakarta_ws_rs.jakarta_ws_rs_api;
 
-import javax.ws.rs.sse.SseEventSource;
+import jakarta.ws.rs.sse.SseEventSource;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
@@ -28,7 +28,7 @@ public class SseEventSourceInnerBuilderTest {
             assertThatThrownBy(() -> SseEventSource.target(null))
                     .isInstanceOf(LinkageError.class)
                     .hasMessageContaining("ClassCastException: attempting to cast")
-                    .hasMessageContaining("javax/ws/rs/sse/SseEventSource$Builder.class");
+                    .hasMessageContaining("jakarta/ws/rs/sse/SseEventSource$Builder.class");
         } finally {
             restoreProviderProperty(previousProvider);
             Thread.currentThread().setContextClassLoader(previousContextClassLoader);

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java
@@ -1,0 +1,108 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package jakarta_ws_rs.jakarta_ws_rs_api;
+
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.sse.SseEventSource;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SseFactoryFinderTest extends SseEventSource.Builder {
+    private static final String SSE_BUILDER_PROPERTY =
+            SseEventSource.Builder.JAXRS_DEFAULT_SSE_BUILDER_PROPERTY;
+    private static final String TEST_SSE_BUILDER_CLASS = SseFactoryFinderTest.class.getName();
+
+    @Test
+    public void targetInstantiatesProviderWithCurrentClassLoaderWhenContextClassLoaderIsNull() {
+        String previousProvider = System.getProperty(SSE_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(SSE_BUILDER_PROPERTY, TEST_SSE_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(null);
+
+            SseEventSource.Builder builder = SseEventSource.target(null);
+
+            assertThat(builder).isInstanceOf(SseFactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void targetInstantiatesProviderWithContextClassLoader() {
+        String previousProvider = System.getProperty(SSE_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(SSE_BUILDER_PROPERTY, TEST_SSE_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(SseFactoryFinderTest.class.getClassLoader());
+
+            SseEventSource.Builder builder = SseEventSource.target(null);
+
+            assertThat(builder).isInstanceOf(SseFactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    @Test
+    public void targetRetriesWithCurrentClassLoaderWhenContextClassLoaderCannotLoadProvider() {
+        String previousProvider = System.getProperty(SSE_BUILDER_PROPERTY);
+        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            System.setProperty(SSE_BUILDER_PROPERTY, TEST_SSE_BUILDER_CLASS);
+            Thread.currentThread().setContextClassLoader(new RejectingClassLoader());
+
+            SseEventSource.Builder builder = SseEventSource.target(null);
+
+            assertThat(builder).isInstanceOf(SseFactoryFinderTest.class);
+        } finally {
+            restoreProviderProperty(previousProvider);
+            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
+        }
+    }
+
+    private static void restoreProviderProperty(String previousProvider) {
+        if (previousProvider == null) {
+            System.clearProperty(SSE_BUILDER_PROPERTY);
+        } else {
+            System.setProperty(SSE_BUILDER_PROPERTY, previousProvider);
+        }
+    }
+
+    private static final class RejectingClassLoader extends ClassLoader {
+        private RejectingClassLoader() {
+            super(null);
+        }
+
+        @Override
+        protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+            throw new ClassNotFoundException(name);
+        }
+    }
+
+    @Override
+    protected SseEventSource.Builder target(WebTarget endpoint) {
+        return this;
+    }
+
+    @Override
+    public SseEventSource.Builder reconnectingEvery(long delay, TimeUnit unit) {
+        return this;
+    }
+
+    @Override
+    public SseEventSource build() {
+        return null;
+    }
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/src/test/java/jakarta_ws_rs/jakarta_ws_rs_api/SseFactoryFinderTest.java
@@ -7,8 +7,8 @@
 package jakarta_ws_rs.jakarta_ws_rs_api;
 
 import java.util.concurrent.TimeUnit;
-import javax.ws.rs.client.WebTarget;
-import javax.ws.rs.sse.SseEventSource;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.sse.SseEventSource;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/user-code-filter.json
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "javax.ws.rs.**"
+    },
+    {
+      "includeClasses" : "jakarta_ws_rs.jakarta_ws_rs_api.**"
+    }
+  ]
+}

--- a/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/user-code-filter.json
+++ b/tests/src/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/user-code-filter.json
@@ -4,7 +4,7 @@
       "excludeClasses" : "**"
     },
     {
-      "includeClasses" : "javax.ws.rs.**"
+      "includeClasses" : "jakarta.ws.rs.**"
     },
     {
       "includeClasses" : "jakarta_ws_rs.jakarta_ws_rs_api.**"

